soc/ite/ec/it51xxx/linker.ld

```diff
@@ -257,6 +257,8 @@ SECTIONS
     /* Claim RAM for ILM mappings; must be 4k-aligned and each mapping is 4k in size */
     SECTION_PROLOGUE(ilm_ram,(NOLOAD),ALIGN(0x1000))
 	{
+		/* On IT51XXX chip, scratch RAM must start at RAM_BASE+0x1000 */
+		. += 0x1000;
```

Comment:
Is this needed? The `ilm_ram` section already includes the `ALIGN(0x1000)` attribute.

Reply:
In soc.c at line 106, we have:
`/* Scratch SRAM0 uses the 4KB based form 0x801000h */`
`gctrl_regs->GCTRL_SCR0BAR = IT51XXX_SEL_SRAM0_BASE_4K;`

If the `GCTRL_SCR0BAR `register is set to 0,  the hardware design does not map this to `RAM_BASE`; instead, it serves a different purpose. Therefore, to use the scratch SRAM at `RAM_BASE+0x1000,` we need to introduce a 0x1000 offset here, resulting in a gap.